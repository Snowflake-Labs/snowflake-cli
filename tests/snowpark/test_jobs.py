from tempfile import NamedTemporaryFile
from unittest import mock


@mock.patch("snowflake.connector.connect")
def test_create_job(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    with NamedTemporaryFile(suffix="yaml") as fh:
        result = runner.invoke(
            [
                "snowpark",
                "jobs",
                "create",
                "--compute-pool",
                "jobName",
                "--spec-path",
                fh.name,
                "--stage",
                "stageValue",
            ]
        )

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_desc_job(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "jobs", "desc", "jobName"])

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_job_status(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "jobs", "status", "jobName"])

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_job_logs(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(
        ["snowpark", "jobs", "logs", "--container-name", "containerName", "jobName"]
    )

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_drop_job(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "jobs", "drop", "cpNameToDrop"])

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot
