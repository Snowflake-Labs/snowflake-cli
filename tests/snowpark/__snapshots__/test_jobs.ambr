# serializer version: 1
# name: test_create_job
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  CREATE STAGE IF NOT EXISTS stageValue;
  
  put file:///var/folders/1b/6450dkj56wx01kj9ls9vj6y00000gn/T/tmpmtmh6582yaml @stageValue/jobs/d41d8cd98f00b204e9800998ecf8427e auto_compress=false OVERWRITE = TRUE;
  
  EXECUTE SERVICE
    COMPUTE_POOL =  jobName
    spec=@stageValue/jobs/d41d8cd98f00b204e9800998ecf8427e/tmpmtmh6582yaml;
  '''
# ---
# name: test_desc_job
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  desc service jobName;
  '''
# ---
# name: test_drop_job
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  call SYSTEM$CANCEL_JOB('cpNameToDrop');
  '''
# ---
# name: test_job_logs
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  call SYSTEM$GET_JOB_LOGS('jobName', 'containerName');
  '''
# ---
# name: test_job_status
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  CALL SYSTEM$GET_JOB_STATUS('jobName');
  '''
# ---
